src/main/java/com/unstoppabledomains/resolution/contracts/uns/ProxyReader.java

```diff
@@ -24,6 +26,17 @@ public String getOwner(BigInteger tokenID) {
         }
     }
 
+    public String[] batchOwners(BigInteger[] tokenIDs) throws Exception {
+            Object[] args = { tokenIDs };
+            BigInteger[] owners = fetchOne("ownerOfForMany", args);
+            String[] convertedOwners = new String[owners.length];
+            for (int i =0; i < owners.length; i++) {
```

Comment:
You can use foreach loop here for convenience: `for (BigInteger owner : owners)`